Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.FileDialog.cs

```diff
@@ -0,0 +1,282 @@
+        // </returns>
+        // [DefaultValue(true)]
+        // public bool ValidateNames { get; set; }
+
+        // <summary>
+        // Gets the custom places collection for this <see cref="T:System.Windows.Forms.FileDialog"/> instance.
+        // </summary>
+        // 
+        // <returns>
+        // The custom places collection for this <see cref="T:System.Windows.Forms.FileDialog"/> instance.
+        // </returns>
+        // public FileDialogCustomPlacesCollection CustomPlaces {get;}
+
+        // <summary>
+        // Gets or sets a value indicating whether this <see cref="T:System.Windows.Forms.FileDialog"/> instance should automatically upgrade appearance and behavior when running on Windows Vista.
+        // </summary>
+        // 
+        // <returns>
+        // true if this <see cref="T:System.Windows.Forms.FileDialog"/> instance should automatically upgrade appearance and behavior when running on Windows Vista; otherwise, false. The default is true.
+        // </returns>
+        // [DefaultValue(true)]
+        // public bool AutoUpgradeEnabled {get; set;}
+
+        // <summary>
+        // Occurs when the user clicks on the Open or Save button on a file dialog box.
+        // </summary>
+        // public event CancelEventHandler FileOk
+
+        // <summary>
+        // Defines the common dialog box hook procedure that is overridden to add specific functionality to the file dialog box.
+        // </summary>
+        // 
+        // <returns>
+        // Returns zero if the default dialog box procedure processes the message; returns a nonzero value if the default dialog box procedure ignores the message.
+        // </returns>
+        // <param name="hWnd">The handle to the dialog box window. </param><param name="msg">The message received by the dialog box. </param><param name="wparam">Additional information about the message. </param><param name="lparam">Additional information about the message. </param>
+        // [SecurityPermission(SecurityAction.LinkDemand, Flags = SecurityPermissionFlag.UnmanagedCode)]
+        // protected override IntPtr HookProc(IntPtr hWnd, int msg, IntPtr wparam, IntPtr lparam)
+
+        // <summary>
+        // Raises the <see cref="E:System.Windows.Forms.FileDialog.FileOk"/> event.
+        // </summary>
+        // <param name="e">A <see cref="T:System.ComponentModel.CancelEventArgs"/> that contains the event data. </param>
+        // protected void OnFileOk(CancelEventArgs e)
+
+        /// <summary>
+        /// Specifies a common dialog box.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// true if the file could be opened; otherwise, false.
+        /// </returns>
+        /// <param name="hWndOwner">A value that represents the window handle of the owner window for the common dialog box. </param>
+        protected override bool RunDialog(IntPtr hWndOwner)
+        {
+            Contract.Requires(!Control.CheckForIllegalCrossThreadCalls ||
```

Comment:
Is there any point in adding a contract to a protected member of a system class? Or would this be enforced for custom subclasses of `FileDialog`?